iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -166,22 +167,33 @@ private void createTableForCTAS(Configuration configuration, Properties serDePro
       serDeProperties.put(InputFormatConfig.PARTITION_SPEC, PartitionSpecParser.toJson(spec));
     }
 
+    // clean up the properties for table creation (so that internal serde props don't become table props)
+    Properties createProps = getCTASTableCreationProperties(serDeProperties);
```

Comment:
If not reused, why not call it inline?

Reply:
I mainly extracted into a function for readability this time, not reuse. But if you think it's better to inline, I can do it as I don't feel strongly about it